pkg/gadgets/trace/exec/tracer/bpf/execsnoop.h

```diff
@@ -24,6 +24,7 @@ struct event {
 	int retval;
 	int args_count;
 	unsigned int args_size;
+	__u8 cwd[MAX_STRING_SIZE];
 	__u8 comm[TASK_COMM_LEN];
 	__u8 args[FULL_MAX_ARGS_ARR];
```

Comment:
Before the patch, the event was 7752 bytes. This adds 4kB to the event. And events sent to a perf ring buffer are rounded up to the next power of 2 size, so that probably makes it 16kB.

```
PerfBufferPages = 64
gadgets.PerfBufferPages*os.Getpagesize() = 256 kB
```
This means the perf per-cpu ring buffer can hold about 16 events before being full.

Should we send a smaller event if `get_cwd` is `false`? Maybe it is ok for this PR, but something to think about if we add more strings.

Reply:
> This adds 4kB to the event.

you're right :( 

> Should we send a smaller event if get_cwd is false? 

I think we should optimize it. I tried to follow the same idea as with args: to use a memory zone to send null terminated strings as str0\0str1\0str2, etc. however I wasn't able to make it work. 

I also tried to send a smaller event when get_cwd is false, but I didn't find an easy way without having to change a lot of the code. I'll try again tomorrow.

Reply:
I handled it by using conditional compilation. I tried to use a constant but it's difficult because the last field of the events structure (args) has already variable length, so modifying the code not to send the cwd is not easy. I think we can use this for now in this way.